File: Data Structures/quiz2/queue.c
#include <stdio.h>
#include <stdlib.h>
#include "node.h"


node* front;
node* rear;
int len=0;

void enqueue(int data){
	node* newNode = malloc(sizeof(node));
	newNode->data = data;
	newNode->next = NULL;

	if(len==0){
		front = newNode;
		rear = newNode;
	}else{
		rear->next = newNode;
		rear = newNode;
	}
	len++;
}

int dequeue(){
	node* trash = front;
	int aux = front->data;
	if(len>0){
		if(len==1){
			front = NULL;
			rear = NULL;
		}else{
			front = front->next;
		}
		free(trash);
		len--;
		return aux;
	}
}

void print_queue(){
	node * aux = front;
	int i;
	for(i=0;i<len;i++){
		printf("%d\n", aux->data);
		aux = aux->next;
	}
}

int main(){

	enqueue(6);
	enqueue(3);
	enqueue(4);
	dequeue();
	print_queue();
	return 0;
}

File: Data Structures/queue.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None


class Queue:
    def __init__(self):
        self.head = None
        self.tail = None

    def enqueue(self, data):
        newNode = Node(data)
        if self.head == None:
            self.head = newNode
            self.tail = newNode
        else:
            self.tail.next = newNode
            self.tail = newNode

    def dequeue(self):
        if self.head != None:
            aux = self.head
            self.head = self.head.next
            return aux.data

    def printQueue(self):
        aux = self.head
        while aux != None:
            print(aux.data)
            aux = aux.next

    def isEmpty(self):
        if self.head == None:
            return True
        else:
            return False

# LL = Queue()

# LL.enqueue(20)
# LL.enqueue(40)
# LL.enqueue(0)
# LL.enqueue(11)

# # LL.dequeue()
# # LL.dequeue()
# # LL.dequeue()
# # LL.dequeue()
# LL.printQueue()
